nrpy/infrastructures/BHaH/CurviBoundaryConditions/CurviBoundaryConditions.py

```diff
@@ -312,37 +325,41 @@ def Cfunction__EigenCoord_set_x0x1x2_inbounds__i0i1i2_inbounds_single_pt(
 
   // Step 3.a: Compute {{x,y,z}}Cart_from_xx, as a
   //           function of i0,i1,i2
-  REAL xCart_from_xx, yCart_from_xx, zCart_from_xx;
+  {type_alias} xCart_from_xx, yCart_from_xx, zCart_from_xx;
   {{
     // xx_to_Cart for Coordinate {rfm_orig.CoordSystem}):
-    REAL xx0 = xx[0][i0];
-    REAL xx1 = xx[1][i1];
-    REAL xx2 = xx[2][i2];
+    {type_alias} xx0 = xx[0][i0];
+    {type_alias} xx1 = xx[1][i1];
+    {type_alias} xx2 = xx[2][i2];
 """
-    body += ccg.c_codegen(
+    tmp_str = ccg.c_codegen(
         [rfm_orig.xx_to_Cart[0], rfm_orig.xx_to_Cart[1], rfm_orig.xx_to_Cart[2]],
         ["xCart_from_xx", "yCart_from_xx", "zCart_from_xx"],
         include_braces=False,
         fp_type=fp_type,
     )
+    tmp_str = tmp_str.replace("REAL", type_alias)
+    body += tmp_str
 
     body += f"""  }}
 
   // Step 3.b: Compute {{x,y,z}}Cart_from_xx_inbounds, as a
   //           function of i0_inbounds,i1_inbounds,i2_inbounds
-  REAL xCart_from_xx_inbounds, yCart_from_xx_inbounds, zCart_from_xx_inbounds;
+  {type_alias} xCart_from_xx_inbounds, yCart_from_xx_inbounds, zCart_from_xx_inbounds;
   {{
     // xx_to_Cart_inbounds for Coordinate {rfm_orig.CoordSystem}):
-    REAL xx0 = xx[0][i0_inbounds];
-    REAL xx1 = xx[1][i1_inbounds];
-    REAL xx2 = xx[2][i2_inbounds];
+    {type_alias} xx0 = xx[0][i0_inbounds];
+    {type_alias} xx1 = xx[1][i1_inbounds];
+    {type_alias} xx2 = xx[2][i2_inbounds];
 """
-    body += ccg.c_codegen(
+    tmp_str = ccg.c_codegen(
         [rfm_orig.xx_to_Cart[0], rfm_orig.xx_to_Cart[1], rfm_orig.xx_to_Cart[2]],
         ["xCart_from_xx_inbounds", "yCart_from_xx_inbounds", "zCart_from_xx_inbounds"],
         include_braces=False,
         fp_type=fp_type,
     )
+    tmp_str = tmp_str.replace("REAL", type_alias)
```

Comment:
This should not be necessary.

Reply:
This has to be done because the tmp variables ccg generates are `REAL` when they need to be, at a minimum, `double`

Reply:
See my comment above & please remove this line